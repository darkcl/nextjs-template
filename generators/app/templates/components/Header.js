import React from 'react';
import PropTypes from 'prop-types';
import urljoin from 'url-join';
import Link from 'next/link';

import withI18N from './withI18N';

const Header = ({ imagePath, t }) => (
  <div className="header">
    <div className="container">
      <Link href="/index" as="/">
        <a><img src={urljoin(imagePath, 'favicon.png')} /></a>
      </Link>
      <span>{t('common:header.title')}</span>
    </div>
  </div>
);

Header.propTypes = {
  t: PropTypes.func,
  imagePath: PropTypes.string,
};

export default withI18N(Header);
