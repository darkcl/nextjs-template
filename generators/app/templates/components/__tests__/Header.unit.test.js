import React from 'react';
import toJson from 'enzyme-to-json';
import { render } from 'enzyme';

import Header from '../Header';

describe('<Header />', () => {
  it('should render correctly with imagePath=/', () => {
    const header = render(<Header imagePath="/"/>);
    expect(toJson(header)).toMatchSnapshot();
  });

  it('should render correctly with imagePath=https://ssl-gumtree.classistatic.com/jobs/job-app/', () => {
    const header = render(<Header imagePath="https://ssl-gumtree.classistatic.com/jobs/job-app/"/>);
    expect(toJson(header)).toMatchSnapshot();
  });
});
