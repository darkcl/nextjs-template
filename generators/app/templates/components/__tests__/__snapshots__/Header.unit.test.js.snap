// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render correctly with imagePath=/ 1`] = `
<div
  class="header"
>
  <div
    class="container"
  >
    <a
      href="/"
    >
      <img
        src="/favicon.png"
      />
    </a>
    <span>
      header.title
    </span>
  </div>
</div>
`;

exports[`<Header /> should render correctly with imagePath=https://ssl-gumtree.classistatic.com/jobs/job-app/ 1`] = `
<div
  class="header"
>
  <div
    class="container"
  >
    <a
      href="/"
    >
      <img
        src="https://ssl-gumtree.classistatic.com/jobs/job-app/favicon.png"
      />
    </a>
    <span>
      header.title
    </span>
  </div>
</div>
`;
